rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java

```diff
@@ -44,35 +45,6 @@
 
 	private static Logger logger = LoggerFactory.getLogger(MavenExecutor.class);
 
-	static class OutputStreamCopyUtil implements Runnable {
-
-		private BufferedReader reader;
-		private PrintStream psOut;
-
-		public OutputStreamCopyUtil(InputStream source, OutputStream out) {
-			reader = new BufferedReader(new InputStreamReader(source));
-			psOut = new PrintStream(out);
-		}
-
-		@Override
-		public void run() {
-			try {
-				String message = "";
-				while (message != null) {
-					psOut.println(message);
-					logger.trace(message);
-					Thread.sleep(100);
```

Comment:
Moved the static inner class to the bottom. Note that waiting 100ms for each line kills the performance.
There is no need to sleep and wait for an interrupt here. If the process producing the output is killed, we should write out all the remaining log entries - and afterwards the process will terminate itself as the input stream ends.